core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -1077,9 +1077,8 @@ public static SqlSingleOperandTypeChecker same(int operandCount,
           SqlTypeFamily.INTEGER);
 
   public static final SqlSingleOperandTypeChecker STRING_INTEGER_OPTIONAL_INTEGER =
-      family(
-          ImmutableList.of(SqlTypeFamily.STRING, SqlTypeFamily.INTEGER,
```

Comment:
These two should have the same meaning or aim to achieve the same effect. Should we fix the first API then?

Reply:
they should, but I don't know how to fix the API, because I don't really understand the rules.
I commented about this in the JIRA ticket
I tried to mark this API Deprecated, but there is still one use I could not remove.

Reply:
I noticed that another PR has made the same changes. If this API is incorrect, I suggest deprecating this API.

Reply:
I need to file another issue; this API is still used by `JSON_QUERY`, which I think is actually not type checked at all because of this.